File: user-service/src/messages/rabitMQ/connection.ts
import amqp  from "amqplib";

let connection;
let channel:any;

export async function connectRabbitMQ(url:string) {
  try {
    connection = await amqp.connect(url);

    channel = await connection.createChannel();

    console.log("RabbitMQ Connected");
  } catch (error) {
    console.error("RabbitMQ Error:", error);
  }
}

export function getChannel() {
  return channel;
}
 

File: task-service/app/rabitmq/connection.py
import pika

class RabitMq:
    _connection = None
    _channel = None
    @classmethod
    def connect(cls, url):
        cls._connection = pika.BlockingConnection(
            pika.URLParameters(url)
            )
        cls._channel = cls._connection.channel()

        print("RabbitMQ Connected")
    
    @classmethod
    def get_channel(cls):
        return cls._channel



